blocklylib-core/src/main/java/com/google/blockly/model/Block.java

```diff
@@ -1325,4 +1347,35 @@ private static boolean containsVariableField(List<Input> inputs) {
         }
         return false;
     }
+
+    private static class EventsCallbackWrapper implements BlocklyController.EventsCallback {
+        private final BlocklyController mController;
+        private final WeakReference<Block> mBlockRef;
+        private final BlocklyController.EventsCallback mCallback;
+
+        EventsCallbackWrapper(
+                @NonNull Block block, @NonNull BlocklyController.EventsCallback callback) {
+            mController = block.getController();
+            mCallback = callback;
```

Comment:
As discussed:
Block points at Callback and Callback Wrapper, Wrapper weak ref to Callback, that way even if the original callback still holds a strong ref to the block they can be garbage collected together.